en/advanced_features/traffic_avoidance_adsb.md

```diff
@@ -94,7 +94,74 @@ The code can be found in `Navigator::check_traffic` ([/src/modules/navigator/nav
 PX4 will also forward the transponder data to a GCS if this has been configured for the MAVLink instance (this is recommended).
 The last 10 Digits of the GUID is displayed as Drone identification.
 
+## Testing/Simulated ADSB Traffic
+
+You can simulate ADS-B traffic for testing.
+Note that this requires that you [Build PX4](../dev_setup/building_px4.md).
+
+:::note
+Simulated ADS-B traffic can trigger real failsafe actions.
+Use with care in real flight!
+:::
+
+To enable this feature:
+
+1. Uncomment the code in `AdsbConflict::run_fake_traffic()`([AdsbConflict.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/adsb/AdsbConflict.cpp#L342C1-L342C1)).
+1. Rebuild and run PX4.
+1. Execute the [`navigator fake_traffic` command](../modules/modules_controller.md#navigator) in the [QGroundControl MAVLink Shell](https://docs.qgroundcontrol.com/master/en/analyze_view/mavlink_console.html) (or some other [PX4 Console or MAVLink shell](../debug/consoles.md), such as the PX4 simulator terminal).
+
+The code in `run_fake_traffic()` is then executed.
+You should see ADS-B warnings in the Console/MAVLink shell, and QGC should also show an ADS-B traffic popup.
+
+By default `run_fake_traffic()` publishes a number of traffic messages (it calls [`AdsbConflict::fake_traffic()`](#fake-traffic-method) to emit each report).
+These simulate ADS-B traffic where there may be a conflict, where there won't be a conflict, as well as spamming the traffic buffer.
+
+::: details Information about the test methods
+
+The relevent methods are defined in [AdsbConflict.cpp](https://github.com/PX4/PX4-Autopilot/blob/main/src/lib/adsb/AdsbConflict.cpp#L342C1-L342C1).
+
+#### `run_fake_traffic()` method
+
+The `run_fake_traffic()` method is run when the `navigator fake_traffic` command is called.
+
+The method calls the `fake_traffic()` method to generate simulated transponder messages around the current vehicle position.
+It passes in the current vehicle position, and information about the simulated traffic, such as callsign, distances, directions, altitude differences, velocities, and emitter types.
+
+The (commented out) code in `run_fake_traffic()` simulates a number of different scenarios, including conflicts and non-conflicts, as well as spamming the traffic buffer.
+
+#### `fake_traffic()` method
+
+`AdsbConflict::fake_traffic()` is called by the [`run_fake_traffic()`](#run-fake-traffic-method) to create individual ADS-B transponder reports.
+
+This takes several parameters, which specify the characteristics of the fake traffic:
```

Comment:
Thanks @farhangnaderi . I've updated slightly to remove the reference info about alt, and to add the description of the fields below.

Note, the PR that updated the code removed the in-source documentation. Can you add it back again please? By making it necessary to modify this code you have made it a public API. That means instructions for its use need to be included. The best place for that is along side the code. 

If you do this, then I will remove the text below, because it won't be needed.